### ISSUE TITLE
Intel 2696v2 error!

### ISSUE THREAD

**Issue by luigivaltulini** [Piker-Alpha/ssdtPRGen.sh#111]
Not possible extract ssdt.
Un know processor model.

Help thanks :)

**Comment by Piker-Alpha** [comment]
I checked http://ark.intel.com/products/series/75291/Intel-Xeon-Processor-E5-2600-v2-Product-Family#@All
and I can't find a E5-2696 v2 processor. The E5-2695 v2 and E5-2697 v2 are both supported by ssdtPRGen.sh so what is the problem?

**Comment by luigivaltulini** [comment]
Piker is this:
http://www.cpu-world.com/CPUs/Xeon/Intel-Xeon%20E5-2696%20v2.html
For now I msrdump have a value of 31.
 Ssdtprgen not recognize the processor, so can I do SSDT.
 this command: ssdtPRGen.sh -w -x 3 0
 Am I doing something wrong?

 The machine runs fast I have a value of about 50 000 in Geekbench.

http://i1351.photobucket.com/albums/p787/gigignopechegno/supermicro/11_zps69522abc.jpg

**Comment by Piker-Alpha** [comment]
1.) Download/use the Beta version of ssdtPRGen.sh
2.) Add the required data to: ~/Library/ssdtPRGen/Data/User Defined.cfg
```
'E5-2696 v2',130,1200,2500,3300,12,24,0,100
```
3.) Run ./ssdtPRGen.sh

**Comment by luigivaltulini** [comment]
yep, work thanks..
I see 6 steep now with msrdump.
i generate ssdt and put in clover  folder.
But after 1 minute go in KP.
is possible patch applecpupower (yosemite) for ivy xeon v2 , whit clover patch not work?

thanks

**Comment by Piker-Alpha** [comment]
I first need to see the output of the script, not the generated SSDR, before I can help you.

You first used ./ssdtPRGen.sh -w 3 -x 0 
Perhaps you need to add these arguments. Please check /vat/log/system.log for errors.

**Comment by luigivaltulini** [comment]
I managed to do this log.
In two cases I used nullcpu, and the system will be stable and perfect.
When removing nullcpu, the system to boot, but just did log 20-30  seconds  on desktop system  restarts. But I can see 12 22 23 24 25 with dmesg.
I tried with the original kext, then I did the patch but nothing, then used the patch of clover but nothing.
Someone tells me that I should use kext 10.9 Acpi .. with 2 CPU xeon.
I can not understand. !!

sorry and thanks again for your time.

https://www.dropbox.com/s/vl8ipf5hv7zqony/Copia%20di%20system.txt?dl=0

**Comment by luigivaltulini** [comment]
and this:
http://i1351.photobucket.com/albums/p787/gigignopechegno/IMG_0421_zps3b70468e.jpg

**Comment by Piker-Alpha** [comment]
Ok so you have two Intel E5-2696 v2 processors with 12 cores and 24 threads. In that case the data in User Defined.cfg need to be changed to:
```
'E5-2696 v2',130,1200,2500,3300,24,48,0,100
```
Have you done that already? If not please change it. Then run: ./ssdtPRGen.sh -p 'E5-2696 v2' -d 3 and copy/paste to output of the script here.

I also se that Clover only sees 12/24 and not 24/48 (thread/cores), but let us get the data and output of the script first.

**Comment by luigivaltulini** [comment]
yep 2 processor , without hyperthreading in bios , why i enable this i have kernel panic, i think kernel not support more 32 core.


System information: Mac OS X 10.10.1 (14B25)
Brandstring 'Intel(R) Xeon(R) CPU E5-2696 v2 @ 2.50GHz'

Checking Sandy Bridge processor data ...
Checking Ivy Bridge processor data ...
Checking Haswell processor data ...
Checking Broadwell processor data ...
Extracting ACPI tables ...Done.4 Name (_HID, "ACPI0004") object(s) found in the DSDT
matchingData:
5b824d9553434b30085f4849440d414350493030303400
5b824d9553434b31085f4849440d414350493030303400
5b824b9953434b32085f4849440d414350493030303400
5b824b9953434b33085f4849440d414350493030303400
Searching for ACPI Processor declarations in Device(SCK0) {}
Two-byte encoding detected (maximum length 0x0FFF/4095)
pkgLengthByte: 0x4d/77
packageLength: 0x95d/2397
scopeLength: 2397
ACPI Processor declaration (C000) found in Device (SCK0) {...} (non ACPI 1.0 compliant)
logicalCore: 0 C000
logicalCore: 1 C002
logicalCore: 2 C004
logicalCore: 3 C006
logicalCore: 4 C008
logicalCore: 5 C00A
logicalCore: 6 C00C
logicalCore: 7 C00E
logicalCore: 8 C010
logicalCore: 9 C012
logicalCore: 10 C014
logicalCore: 11 C016
Warning: only 12 of 24 ACPI Processor declarations found in Device(SCK0) {}gProcessorStartIndex: 12
gLogicalCPUs        : 24
gProcessorNames     : 24
Searching for 12 additional Processor declaration ...
Searching for ACPI Processor declarations in Device(SCK1) {}
Two-byte encoding detected (maximum length 0x0FFF/4095)
pkgLengthByte: 0x4d/77
packageLength: 0x95d/2397
scopeLength: 2397
ACPI Processor declaration (C100) found in Device (SCK1) {...} (non ACPI 1.0 compliant)
logicalCore: 12 C100
logicalCore: 13 C102
logicalCore: 14 C104
logicalCore: 15 C106
logicalCore: 16 C108
logicalCore: 17 C10A
logicalCore: 18 C10C
logicalCore: 19 C10E
logicalCore: 20 C110
logicalCore: 21 C112
logicalCore: 22 C114
logicalCore: 23 C116
Extracting ACPI tables ...Done.Generating ssdt.dsl for a 'MacPro6,1' with board-id [Mac-F60DEB81FF30ACF6]
Ivy Bridge Core E5-2696 v2 processor [0x306E4] setup [0x0a01]
E5-2696 v2,77,0,-1,,12,24
Number logical CPU's: 24 (Core Frequency: 2499 MHz)
Number of Turbo States: 0
Number of P-States: 9 (1600-2499 MHz)
targetCStates: 13
Adding C1
Adding C3
Adding C6
targetCStates: 13
Adding C1
Adding C3
Adding C6
targetCStates: 13
Adding C1
Adding C3
Adding C6
targetCStates: 13
Adding C1
Adding C3
Adding C6
Injected C-States for C000 (C1,C3,C6)
Injected C-States for C002 (C1,C3,C6)
Warning: 'system-type' may be set improperly (1 instead of 3)

Intel ACPI Component Architecture
ASL Optimizing Compiler version 20140926-64 [Nov  6 2014]
Copyright (c) 2000 - 2014 Intel Corporation

ASL Input:     /Users/RED/Library/ssdtPRGen/ssdt.dsl - 640 lines, 18490 bytes, 171 keywords
AML Output:    /Users/RED/Library/ssdtPRGen/ssdt.aml - 4330 bytes, 78 named objects, 93 executable opcodes

Compilation complete. 0 Errors, 0 Warnings, 0 Remarks, 0 Optimizations

**Comment by Piker-Alpha** [comment]
Ok one step closer, but the script can't seem to find: ~/Library/ssdtPRGen/Data/User Defined.cfg (with a space in the name). See my previous comments about adding/editing this file. In short. You don't want to see this: "E5-2696 v2,77,0,-1,,12,24" text, which means that the data is incomplete, but what you want is this text:

Checking user defined processor data ...
Processor matched!

**Comment by luigivaltulini** [comment]
ok  i have this:

Checking Sandy Bridge processor data ...
Checking Ivy Bridge processor data ...
Extracting ACPI tables ...Done.4 Name (_HID, "ACPI0004") object(s) found in the DSDT
matchingData:
5b824d9553434b30085f4849440d414350493030303400
5b824d9553434b31085f4849440d414350493030303400
5b824b9953434b32085f4849440d414350493030303400
5b824b9953434b33085f4849440d414350493030303400
Searching for ACPI Processor declarations in Device(SCK0) {}
Two-byte encoding detected (maximum length 0x0FFF/4095)
pkgLengthByte: 0x4d/77
packageLength: 0x95d/2397
scopeLength: 2397
ACPI Processor declaration (C000) found in Device (SCK0) {...} (non ACPI 1.0 compliant)
logicalCore: 0 C000
logicalCore: 1 C002
logicalCore: 2 C004
logicalCore: 3 C006
logicalCore: 4 C008
logicalCore: 5 C00A
logicalCore: 6 C00C
logicalCore: 7 C00E
logicalCore: 8 C010
logicalCore: 9 C012
logicalCore: 10 C014
logicalCore: 11 C016
Warning: only 12 of 24 ACPI Processor declarations found in Device(SCK0) {}gProcessorStartIndex: 12
gLogicalCPUs        : 24
gProcessorNames     : 24
Searching for 12 additional Processor declaration ...
Searching for ACPI Processor declarations in Device(SCK1) {}
Two-byte encoding detected (maximum length 0x0FFF/4095)
pkgLengthByte: 0x4d/77
packageLength: 0x95d/2397
scopeLength: 2397
ACPI Processor declaration (C100) found in Device (SCK1) {...} (non ACPI 1.0 compliant)
logicalCore: 12 C100
logicalCore: 13 C102
logicalCore: 14 C104
logicalCore: 15 C106
logicalCore: 16 C108
logicalCore: 17 C10A
logicalCore: 18 C10C
logicalCore: 19 C10E
logicalCore: 20 C110
logicalCore: 21 C112
logicalCore: 22 C114
logicalCore: 23 C116
Extracting ACPI tables ...Done.Generating ssdt.dsl for a 'MacPro6,1' with board-id [Mac-F60DEB81FF30ACF6]
Unknown Core E5-2696 v2 processor [0x306E4] setup [0x0a01]
Processor matched!
With a maximum TDP of 130 Watt, as specified by Intel
Number logical CPU's: 24 (Core Frequency: 2500 MHz)
Number of Turbo States: 8 (2600-3300 MHz)
Number of P-States: 22 (1200-3300 MHz)
targetCStates: 13
Adding C1
Adding C3
Adding C6
targetCStates: 13
Adding C1
Adding C3
Adding C6
Injected C-States for C000 (C1,C3,C6)
Warning: 'board-id' [Mac-F60DEB81FF30ACF6] is not supported by Unknown power management
Warning: 'system-type' may be set improperly (1 instead of 0)

Intel ACPI Component Architecture
ASL Optimizing Compiler version 20140926-64 [Nov  6 2014]
Copyright (c) 2000 - 2014 Intel Corporation

**Comment by luigivaltulini** [comment]
i tried with this , put in clover, but i see only state 12 and 25 and i have KP after 20 second on desktop.

**Comment by Piker-Alpha** [comment]
Where did you add the data? Is that ~/Library/ssdtPRGen/Data/Ivy Bridge.cfg because I see this:
```
Checking Sandy Bridge processor data ...
Checking Ivy Bridge processor data …
```
It should not get there. Not if you add the data to: ~/Library/ssdtPRGen/Data/User Defined.cfg. Then it should also show the line
```
Checking user defined processor data …
```
I also see this
```
Unknown Core
```
This may be due to the fact that you added the data to the other file. You could try to add the -c 1 argument because that usually solves this problem. Still. It would have been nice if it picked up the value automatically.

Also. Since I don't have the generated SSDT you should check if you see a Method (_DSM) {} in it or not. If not then the -c 1 argument should also solve that. Please give it a try and let me know how it goes.

p.s. You did add all cores/threads in the data or not? Note that using the -l 48 argument could help.

**Comment by luigivaltulini** [comment]
Pike, not have this directory . ~/Library/ssdtPRGen/Data/User Defined.cfg

yes i put data in ivy bridge.

**Comment by Piker-Alpha** [comment]
Right. That explains it. I thought to have asked you to download the Beta. Yup. See my second reply. No problem, but that is why I am getting a little confused. Anyway. Did you add the -c 1 argument already? Did that solve the Unknown Core problem? Also try -l 48 (lower case L).

**Comment by luigivaltulini** [comment]
this is my ssdt

https://www.dropbox.com/s/js766e8p1fask3o/SSDT.aml?dl=0

i correct cpu why is 3500 not 3300 and tpd is 120 not 130  from cpu-z.
And my system is 24 core logic, mot possible activate hyperthreathing why i have kernel panic.
if you want i try compile with option -l 48 ?

thanks :)

**Comment by luigivaltulini** [comment]
Yep core problem whit c1 now is ivy , but sharem is  same after desktop 30 second go in kp:(

**Comment by Piker-Alpha** [comment]
When do you get a KP with hyper threading enabled in the BIOS? Is that at boot time before it starts uo properly?

**Comment by luigivaltulini** [comment]
Yep  i have kp

**Comment by Piker-Alpha** [comment]
At boot time? Then try cpus=24 and see when it starts to KP.

**Comment by luigivaltulini** [comment]
I turn off in bios hyper (total 24), the system booting, enter the desktop and after 30 seconds the system will go into kp.

and this capturede on fly:
http://i1351.photobucket.com/albums/p787/gigignopechegno/Schermata2015-01-12alle015243_zpsc6f061e7.png

**Comment by luigivaltulini** [comment]
whit nullcpu kext all work, without KP, but the state is only 25

**Comment by Piker-Alpha** [comment]
Output is looking fine for 24 logical cores. Have you tried to use -l 48 already?

And yes. That kext basically disables power management, but that is not what you want. I also see that someone told you (who by the way?) to use the AppleACPIPlatform.kext from 10.9 and I am wondering if you are using it or not.

**Comment by luigivaltulini** [comment]
Yep, not work o state with -l 48,
no I do not use the kext 10.9, I can not. find or to download it.

http://www.insanelymac.com/forum/topic/302310-sandy-bridge-e-ivy-bridge-e-power-management-1010-yosemite/

hard to go without a guide.

**Comment by Piker-Alpha** [comment]
I added a link to the official Apple 10.9.1 DMG in that thread. Use pacifist to extract the files.

**Comment by luigivaltulini** [comment]
zhanks :)
but this update not have this 2 kext, i extract with pacifist. :(

**Comment by Piker-Alpha** [comment]
Sorry. There is no 10.9.1 Combo update. You need to use the 10.9.2 Combo update DMG. Link updated.

**Comment by luigivaltulini** [comment]
the problem is 10.9.0 or 10.9.1 not 10.9.2 is difficult to find :(

**Comment by Piker-Alpha** [comment]
You were told to use the files from OS X 10.9/0.9.1 and we have yet to find out if using the kext from 10.9.2 works or not. Just give it a try. Also restore the original AppleIntelCPUPowerManagement.kext from Yosemite.
